### ISSUE TITLE
0.75.4: Local image not displayed by <Image /> when extension is not explicit

### ISSUE THREAD

**Issue by fbp93** [facebook/react-native#46870]
### Description

My app is downloading differente pngs saving them without an explicit extension in order to show them later through the `<Image />` react native component(you can find an example [here](https://github.com/user-attachments/files/17284059/test.zip)). This was working fine, but starting from React 0.73.0  the image is not displayed correctly.
Adding an explicit extension to the file (e.g. manually adding the .png) seems fixing it, but my app have several images already downloaded so it's not so easy to handle existing users. IMO it's still an unwanted issue that should be fixed considering it's a breaking change.

### Steps to reproduce

1. Install the application from the [reproducer](https://github.com/fbp93/RN75/tree/master)
2. Run the app, check the directory path in the logs
3. copy the testing [images](https://github.com/user-attachments/files/17284380/images.zip) in the directory at that path
4. update the `path` var in App.tsx with the path of the directory
3. Run the app and see that only one image (the one with the explicit extension) is displayied

### React Native Version

0.75.4

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.6.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 133.64 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.12.1
    path: ~/Library/Caches/fnm_multishells/60602_1728334021609/bin/node
  Yarn:
    version: 3.6.4
    path: /usr/local/bin/yarn
  npm:
    version: 8.19.2
    path: ~/Library/Caches/fnm_multishells/60602_1728334021609/bin/npm
  Watchman:
    version: 2024.09.23.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods: Not Found
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 16.0/16A242
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 3.0.0
    path: /Users/work/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

(node:68279) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
```

### Stacktrace or Logs

```text
No stacktrace
```

### Reproducer

https://github.com/fbp93/RN75/tree/master

**Comment by fbp93** [comment]
@cipolleschi 🙏

**Comment by cipolleschi** [comment]
@fbp93 thanks for the issue.
IIUC, this got broken in 0.73, right? From which version did you migrate? 

To set expectation, this might take a while for us, given that 73 is almost going out of the support window and we almost released already 3 major versions.

**Comment by fbp93** [comment]
> @fbp93 thanks for the issue. IIUC, this got broken in 0.73, right? From which version did you migrate?
> 
> To set expectation, this might take a while for us, given that 73 is almost going out of the support window and we almost released already 3 major versions.

I'm moving from 0.71.16 to 0.75.4. I tested different versions and I found out that the issue started to happen in 0.73.0 (the 0.72.17 is working fine).

> To set expectation, this might take a while for us, given that 73 is almost going out of the support window and we almost released already 3 major versions.
Even if the issue started in 0.73 it's still happening in 0.75 (I didn't tested 0.76 yet). Do you think it's gonna take a while even if it's still a bug in the latest release?

**Comment by cipolleschi** [comment]
Yes, probably. We are currently a bit swamped for 0.76 and having to investigate when this happened and how to properly fix this might take a while. Unless you guys had some time to look at what changed between 0.72 and 0.73 in the Image component.

Also, is this happening on both architecture or only on the Old Architecture?

**Comment by fbp93** [comment]
> Unless you guys had some time to look at what changed between 0.72 and 0.73 in the Image component.
I'm checking all files [here](https://github.com/facebook/react-native/tree/main/packages/react-native/Libraries/Image) but I don't see major changes 🤔 Is there any other place where I should have a look related to the image component?

> Also, is this happening on both architecture or only on the Old Architecture?
i m currently testing on the old architecture

**Comment by cipolleschi** [comment]
> I'm checking all files [here](https://github.com/facebook/react-native/tree/main/packages/react-native/Libraries/Image?rgh-link-date=2024-10-08T16%3A14%3A04Z) but I don't see major changes 🤔 Is there any other place where I should have a look related to the image component?

I don't think.. But you can also debug what's happening in the [`RCTImageLoader`](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Image/RCTImageLoader.mm)! Breakpoints works great in Xcode.

> i m currently testing on the old architecture

If you create a reproducer with this [template](https://github.com/react-native-community/reproducer-react-native), could you test also onn the New Architecture, please?

**Comment by fbp93** [comment]
@cipolleschi Also, is there a way to install a specific commit of RN? I wanna find where it started to break,  I'm trying with 
```
 "dependencies": {
    "react-native": "git+https://github.com/facebook/react-native.git#1eb4bf0d3af1fe267f5de6bf4bbe1e65d41fea1b",
```
but packages are not installed correctly so the pod install doesn't work.

**Comment by cipolleschi** [comment]
no, it's not possible easily. You should package react-native as a proper NPM package. 
Your best approximation is using [nightlies](https://www.npmjs.com/package/react-native?activeTab=versions)

**Comment by fbp93** [comment]
I tried a bit but it's hard without testing the single commits and having some context. Considering I'm planning to ship this in around 1 month, do you think will there any chance that any of you will have some time to have a look in this timeframe or do you think you ll be busy for more time? Just to understand how much effort or alternative I should put on this. 🙏

**Comment by zhongwuzw** [comment]
@fbp93 Hi, I made a PR to fix this issue, is this #46971 work for you?

**Comment by fbp93** [comment]
> @fbp93 Hi, I made a PR to fix this issue, is this #46971 work for you?

@zhongwuzw I ll test it better in the coming days but it seems working fine thanks ❤️

**Comment by AliRehman7141** [comment]
I'm also facing this issue, static images are not visible but server images are working fine.
```
"react": "18.3.1",
"react-native": "0.75.3",
```

**Comment by fbp93** [comment]
@zhongwuzw are you planning to release a patch this week with the fix?

**Comment by zhongwuzw** [comment]
@fbp93 Hi, here are the steps I suggest:

The pull request [#46971](https://github.com/facebook/react-native/pull/46971) should first be reviewed and merged by the React Native team. Perhaps @cipolleschi could assist with the review?
Once it's merged, @fbp93, you can request to cherry-pick this commit by submitting a new issue at [React Native Releases](https://github.com/reactwg/react-native-releases/issues/new/choose).

**Comment by fbp93** [comment]
@cipolleschi LMK when you have some time to release the fix, can we also release the patch for the 0.73.x and 0.74.x that are affected 🙏 ?

**Comment by cipolleschi** [comment]
@zhongwuzw thanks for looking into this. I'm a bit confused because the code you modified touches a section of the code that is 5 years old.
If the code has stopped working recently, the change should be somewhere else. 🤔

**Comment by zhongwuzw** [comment]
That's interesting! I reviewed the code, and I didn't notice any other related changes that could affect image loading.

@fbp93, could you let us know which version of React Native does not have this issue so that I can find wether any other code affects this? Is it v0.72.x?

**Comment by cipolleschi** [comment]
Heads up that I had to revert the fix, because it was breaking an internal app. I hope to be able to restore it and fix the internal app this week.

**Comment by fbp93** [comment]
👋 This has never been restored, isn't it?

**Comment by cipolleschi** [comment]
Yeah, my bad, sorry.
I've been a bit swamped by other tasks.
This PR should fix all the cases: https://github.com/facebook/react-native/pull/48888

I need to test it a bit further before being able to land this.

**Comment by cipolleschi** [comment]
I was also looking at your reproducer, but as soon as i copy both the `test` image in the simulator folder 
```
~/Library/Developer/CoreSimulator/Devices/2931283F-FB43-4082-8E82-075F0DB6286F/data/Containers/Bundle/Application/1E737D06-C41C-4B70-B9CB-0E5761167C17
```
It worked fine without any change in the codebase... 😕 

<img src="https://github.com/user-attachments/assets/1a2faa7e-7871-495e-a5cb-2913ce075634" height="320"/>
